src/com/amplitude/api/DatabaseHelper.java

```diff
@@ -96,6 +96,12 @@ public void onCreate(SQLiteDatabase db) {
         // lifetime of the table, even if rows get removed
         db.execSQL(CREATE_EVENTS_TABLE);
         db.execSQL(CREATE_IDENTIFYS_TABLE);
+
+        if (databaseResetListener != null && callResetListenerOnDatabaseReset) {
+            callResetListenerOnDatabaseReset = false;  // guards against stack overflow
+            databaseResetListener.onDatabaseReset(db);
```

Comment:
if onDatabaseReset fails, callResetListenerOnDatabaseReset will never get reset

Reply:
kk putting it in a try catch and resetting flag in finally